### ISSUE TITLE
de.lessvoid.nifty.batch.spi not in 1.4

### ISSUE THREAD

**Issue by toolforger** [nifty-gui/nifty-gui#102]
Which means I had to revert to using 1.3 because a library I'm using started calling that interface, forcing me to backport the various fixes I submitted for 1.4 to keep my application running :-(
And guess what, 1.3 initializes things in a different order and now I'm getting NPEs because the `element` member isn't initialized in time. And I don't know what change influenced that.
This all just threw me back by a week. Can't say I'm happy about that.

**Comment by void256** [comment]
Joachim,

I'm sorry that this is causing you pain.

The batch renderer will soon be in 1.4. I just didn't want to develop it in parallel in two branches. So it will be in 1.4 once it is stable enough in 1.3.3. There are only two minor issues remaining (a rendering error on android and a missing API feature) so that should be soon.

I think the library you refer to is the latest nightly build of jMonkeyEngine since this is the only lib I can think of that uses this feature already. The feature itself is optional but I guess it's not binary compatible with 1.4 anymore because of the missing interfaces and classes in Nifty 1.4 that have been introduced in 1.3.3.

However, 1.3.3 is not released yet because of the two issues I mentioned above. Therefore the jMonkeyEngine nightly build uses a nightly build of Nifty 1.3.3-SNAPSHOT internally. The only reason for that somewhat fragile situation is to allow very early testing of the batched renderer in jME. Which looks well so far.

The whole situation will improve in the near future (a couple of weeks in the worst case). 1.3.3 will be released and the batch renderer will be ported to 1.4.

void

**Comment by toolforger** [comment]
In other words, you just pulled out an essential part of my software stack out of my project and don't plan to put it back in for several weeks.
Fine.
I'll try returning to 1.3.3, with the risk of losing another couple of weeks.
This will make it too expensive for me to contribute to any version of Nifty other than what's currently in use in JME, so I guess I'll stop working on and with 1.4.

**Comment by void256** [comment]
The new batched renderer stuff has now been merged into 1.4.

**Comment by toolforger** [comment]
Heh. Half a day after I started transitioning to another gui lib.
I had some stuff I could do while JME/Nifty/myapp was broken, so I didn't lose those two weeks in the end. However, you didn't say something like that won't happen again (fair enough), and I'd likely be in real trouble next time it does, so I guess my best option is to continue the ongoing gui lib migration and drop Nifty.
